src/librustc/middle/subst.rs

```diff
@@ -555,6 +555,11 @@ impl<'a,T> Iterator for EnumeratedItems<'a,T> {
             None
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let size = self.vec.as_slice().len();
+        (size, Some(size))
```

Comment:
No, it doesn't. The underlying vector is used in its entirety. If you check `VecPerParamSpace::limits()` one can see how it is split into three consecutive ranges.

(We discussed this on IRC, I'm just adding this comment for reference.)